kyuubi-common/src/main/scala/org/apache/kyuubi/service/TempFileService.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.service
+
+import java.nio.file.{Path, Paths}
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicLong
+
+import com.google.common.cache.{Cache, CacheBuilder, RemovalNotification}
+
+import org.apache.kyuubi.Utils
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.service.TempFileService.tempFileCounter
+import org.apache.kyuubi.util.{TempFileCleanupUtils, ThreadUtils}
+
+class TempFileService(name: String) extends AbstractService(name) {
+  def this() = this(classOf[TempFileService].getSimpleName)
+
+  final private var expiringFiles: Cache[String, String] = _
+  private lazy val cleanupScheduler =
+    ThreadUtils.newDaemonSingleThreadScheduledExecutor(s"temp-file-cleanup-scheduler")
+
+  override def initialize(conf: KyuubiConf): Unit = {
+    super.initialize(conf)
+    val interval = conf.get(KyuubiConf.SERVER_STALE_FILE_EXPIRATION_INTERVAL)
+    expiringFiles = CacheBuilder.newBuilder()
+      .expireAfterWrite(interval, TimeUnit.MILLISECONDS)
+      .removalListener((notification: RemovalNotification[String, String]) => {
+        val pathStr = notification.getValue
+        cleanupFilePath(pathStr)
```

Comment:
add a info log

Reply:
I would prefer a debug log here. It's maybe noisy to have a info log, as every session log root path will be cleaned-up by expiration.

Added the debug log on removing expired file here.